src/stream/src/executor/temporal_filter.rs

```diff
@@ -0,0 +1,466 @@
+            let no_longer_owned_vnodes =
+                Bitmap::bit_saturate_subtract(prev_vnode_bitmap, curr_vnode_bitmap);
+            self.buffer.retain(|(_, pk), _| {
+                let vnode = self.state_table.compute_vnode(pk);
+                !no_longer_owned_vnodes.is_set(vnode as _)
+            });
+        }
+
+        let values_per_vnode =
+            get_rowstreams_by_new_vnodes(prev_vnode_bitmap, curr_vnode_bitmap, &self.state_table)
+                .await?;
+        if !values_per_vnode.is_empty() {
+            let mut stream = select_all(values_per_vnode);
+            while let Some(storage_result) = stream.next().await {
+                // Insert the data into buffer.
+                let row = storage_result?.into_owned();
+                let timestamp = row[self.time_col_idx].as_ref().unwrap().clone();
+                let pk = row.by_indices(&self.pk_indices);
+                self.buffer.insert((timestamp, pk), row);
+            }
+        }
+        Ok(())
+    }
+
+    fn sync(
+        &mut self,
+        lower_condition: &BoxedExpression,
+    ) -> StreamExecutorResult<Vec<StreamChunk>> {
+        let identity = self.identity.clone();
+        let mut ret = vec![];
+        let mut data_chunk_builder =
+            DataChunkBuilder::new(self.schema().data_types(), self.chunk_size);
+        while let Some(entry) = self.buffer.first_entry() {
+            if lower_condition
+                .eval_row_infallible(entry.get(), |err| self.ctx.on_compute_error(err, &identity))
+                .map_or(false, |scalar| *scalar.as_bool())
+            {
+                break;
+            } else {
+                // Remove the record from buffer.
+                let row = entry.remove();
+
+                // If there is higher condition, we need to check existence before yield.
+                if let Some(data_chunk) =
+                    data_chunk_builder.append_one_row_from_datums(row.values())
+                {
+                    // When the chunk size reaches its maximum, we construct a
+                    // stream chunk
+                    let ops = vec![Op::Delete; data_chunk.capacity()];
+                    ret.push(StreamChunk::from_parts(ops, data_chunk));
+                }
+
+                self.state_table.delete(row);
+            }
+        }
+        if let Some(data_chunk) = data_chunk_builder.consume_all() {
+            let ops = vec![Op::Delete; data_chunk.capacity()];
+            ret.push(StreamChunk::from_parts(ops, data_chunk));
+        }
+        Ok(ret)
+    }
+
+    #[try_stream(ok = Message, error = StreamExecutorError)]
+    async fn into_stream(mut self) {
+        let time_col_data_type = self.schema().fields()[self.time_col_idx].data_type();
+        let window_size = *self.window_size.as_interval();
+        let dynamic_cond = move |time_millis: u64| -> Result<BoxedExpression, ExprError> {
+            new_binary_expr(
+                Type::GreaterThanOrEqual,
```

Comment:
I guess we can support both GTOE and GT in the future (minor detail)

Reply:
> I guess we can support both GTOE and GT in the future (minor detail)

sure